engine/flink/components/base-unbounded/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/aggregates.scala

```diff
@@ -69,19 +74,38 @@ object aggregates {
 
   }
 
+  object MedianAggregator extends Aggregator with LargeFloatingNumberAggregate {
+
+    override type Aggregate = ListBuffer[Number]
+
+    override type Element = Number
+
+    override def zero: Aggregate = ListBuffer.empty
+
+    override def addElement(el: Element, agg: Aggregate): Aggregate = if (el == null) agg else agg.addOne(el)
+
+    override def mergeAggregates(agg1: Aggregate, agg2: Aggregate): Aggregate = agg1 ++ agg2
+
+    override def result(finalAggregate: Aggregate): AnyRef = MedianHelper.calculateMedian(finalAggregate.toList)
+
+    override def computeStoredType(input: TypingResult): Validated[String, TypingResult] = Valid(
+      Typed.genericTypeClass[ListBuffer[_]](List(input))
+    )
+
+  }
+
   object ListAggregator extends Aggregator {
 
-    override type Aggregate = List[AnyRef]
+    override type Aggregate = ListBuffer[AnyRef]
```

Comment:
I'm afraid it can break state compatibility in Flink jobs. It would be more safe to extract these changes to a separate PR and discuss it.